tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2409,35 +2409,26 @@ SEC("raw_tracepoint/sched_process_exit")
 int tracepoint__sched__sched_process_exit(struct bpf_raw_tracepoint_args *ctx)
 {
     context_t context = init_context();
-    if (!should_trace(&context)) {
-        // Note: we need to remove the container id here as we always add it to the map in cgroup_attach_task event.
-        bpf_map_delete_elem(&pid_to_cont_id_map, &context.host_tid);
-        bpf_map_delete_elem(&process_tree_map, &context.host_pid);
-        return 0;
-    }
+
+    // Remove this pid from all maps
+    bpf_map_delete_elem(&pid_to_cont_id_map, &context.host_tid);
     bpf_map_delete_elem(&process_tree_map, &context.host_pid);
+    bpf_map_delete_elem(&traced_pids_map, &context.host_tid);
+    bpf_map_delete_elem(&new_pids_map, &context.host_tid);
+
```

Comment:
- pid_to_cont_id_map is written by:
   - raw_tracepoint/sched_process_fork (with child's pid if existing container_id for the parent)
   - raw_tracepoint/cgroup_attach_task (during a cgroup attachment if prefix docker-XXX)

- process_tree_map is written by:
   - raw_tracepoint/sched_process_fork (with child's pid if parent has an entry)

- traced_pids_map is written by:
   - raw_tracepoint/sys_enter for SYS_EXECVE and SYS_EXECVEAT (child - after should_trace if parent is traced)
   - raw_tracepoint/sched_process_fork (child - after should_trace if parent is traced)

- new_pids_map is written by:
   - raw_tracepoint/sys_enter for SYS_EXECVE and SYS_EXECVEAT (child - if NEW_PID_FILTER)
   - raw_tracepoint/sched_process_fork (child - if NEW_PID_FILTER)

LGTM